README.md

```diff
@@ -140,12 +162,12 @@ Then write some unit tests that exercise your function, making sure that it
 recognizes valid sentences and rejects invalid ones.
 
 The answer should appear in one or more source files in the same directory as
-this file. If I need to do anything more that type a single command to run your
+this file. If I need to do anything more than type a single command to run your
 code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+python fsm.py
```

Comment:
You used a table,  and you included a nice set of tests, both positive and negative. The only problem was that ambiguity you introduced in the table, which lead to you having to write:

~~~ py
                if len(next)==1: state = next[0]
                else:
                    if words[index + 1]=='dog' or words[index + 1]=='cat': 
                        state = next[1]
                    else: 
                        state = next[0]
~~~

which is clearly not clean.

# -2

(I was looking for a table that looked more like

~~~ py
states = {
  0:  { "the":    1 },
  1:  { "lazy":   2, "smelly":  2, "dog": 3, "cat": 3 },
  2:  { "lazy":   2, "smelly":  2, "dog": 3, "cat": 3 },
  3:  { "ate":    4, "ran":     4 },
  4:  { "slowly": 5, "noisily": 5, "EOI": OK },
  5:  { "EOI":    OK }
}

def getNextState(currentState, nextWord):
    return states.get(currentState,{}).get(nextWord, ERROR)

def stateMachine(sentence):
    return reduce(getNextState, sentence, 0) == OK
~~~